cardano-cli/src/Cardano/CLI/Shelley/Run/Query.hs

```diff
@@ -122,48 +144,39 @@ writeProtocolParameters mOutFile pparams =
         LBS.writeFile fpath (encodePretty pparams)
 
 runQueryTip
-  :: Network
+  :: NetworkId
   -> Maybe OutputFile
   -> ExceptT ShelleyQueryCmdError IO ()
 runQueryTip network mOutFile = do
-  sockPath <- firstExceptT ShelleyQueryEnvVarSocketErr readEnvSocketPath
-  let ptclClientInfo = pClientInfoCodecConfig
-                     . protocolClientInfo
-                     $ mkNodeClientProtocolShelley
-  tip <- liftIO $ withIOManager $ \iomgr ->
-    getLocalTip iomgr ptclClientInfo network sockPath
+  SocketPath sockPath <- firstExceptT ShelleyQueryEnvVarSocketErr readEnvSocketPath
+  let ptcl = mkNodeClientProtocolShelley
+  tip <- liftIO $ getLocalTip sockPath network ptcl
   case mOutFile of
     Just (OutputFile fpath) -> liftIO . LBS.writeFile fpath $ encodePretty tip
     Nothing -> liftIO $ LBS.putStrLn (encodePretty tip)
 
 
 runQueryUTxO
   :: QueryFilter
-  -> Network
+  -> NetworkId
   -> Maybe OutputFile
   -> ExceptT ShelleyQueryCmdError IO ()
 runQueryUTxO qfilter network mOutFile = do
-  sockPath <- firstExceptT ShelleyQueryEnvVarSocketErr readEnvSocketPath
-  let ptclClientInfo = pClientInfoCodecConfig
-                     . protocolClientInfo
-                     $ mkNodeClientProtocolShelley
-  tip <- liftIO $ withIOManager $ \iomgr ->
-            getLocalTip iomgr ptclClientInfo network sockPath
+  SocketPath sockPath <- firstExceptT ShelleyQueryEnvVarSocketErr readEnvSocketPath
```

Comment:
Looks like these three lines can be factored out (DRY).

Reply:
Yeah. In fact later I want to go one better and get the tip within the same protocol session with the node rather than two.